x-pack/plugins/security/server/lib/authentication/__tests__/authenticator.js

```diff
@@ -153,6 +154,10 @@ describe('Authenticator', () => {
       const user = { username: 'user' };
       const systemAPIRequest = requestFixture({ headers: { authorization: 'Basic xxx' } });
```

Comment:
I find this test hard to follow now that we're setting both basic auth headers, which shouldn't be creating a session alone, and also setting the login attempt. Are the basic authorization headers being used in this test except to disambiguate between which requests are a "system API request" vs a "not system API request"?

Reply:
I found all of these tests hard to follow, so let me give this some thought and see if I can propose a more clear way to break this down.

Reply:
I pushed 7280f9fd30a0b995cea9c17e56d8c9048b571dd9 which should make this test more clear. I'll push a change for your other similar comment independently.

Reply:
https://github.com/elastic/kibana/commit/7280f9fd30a0b995cea9c17e56d8c9048b571dd9 is great! Thanks for improving these tests.